src/java/org/apache/cassandra/io/sstable/metadata/StatsMetadata.java

```diff
@@ -536,13 +537,9 @@ public StatsMetadata deserialize(Version version, DataInputPlus in) throws IOExc
 
             if (version.hasZeroCopyMetadata() && in.readByte() != 0)
             {
-                logger.warn("Ignoring zero copy metadata from {} as it is not supported", in);
-                in.skipBytes(Long.BYTES +                   // start offset
-                             Long.BYTES +                   // end offset
-                             Integer.BYTES +                // chunk size
-                             Long.BYTES);                   // estimated keys
-                ByteBufferUtil.readWithShortLength(in);     // first key
-                ByteBufferUtil.readWithShortLength(in);     // last key
+                throw new UnsupportedSSTableException(String.format("Found zero copy metadata in %s. It is not supported for the SSTables to be created this way.", in),
```

Comment:
The message looks wrong -- why "created"?

How about "Found DSE zero copy metadata in %s. Files copied over using partial zero-copy streaming in DSE are not currently supported." (DSE 6.8 and C* 4.0 both have ZCS but they are different: C* only copies whole files, DSE can do both whole and partial, and it needs the zero-copy metadata for the latter kind.)

Reply:
sorry, it was a mistake, thanks for exact wording